hummingbot/market/dolomite/prepareOrderTest

```diff
@@ -0,0 +1,171 @@
+
+import binascii 
+import aiohttp
+import time 
+import asyncio
+import json
+from collections import (
+    deque,
+    OrderedDict
+)
+import logging
+import math
+import time
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional
+)
+from decimal import Decimal
+from web3 import Web3
+
+
+
+import requests 
+
+
+#Dolomite post requests must be in JSON
+#Make sure endpoint attached to routes doesn't include extra backslash 
+#Make sure when retrieving data ["data"] attribute is accessed first when relevant as a prefix 
+#Headers?
+
+infura_url = "https://mainnet.infura.io/v3/d2de8f9c500e4a5c9761ef1ffdd98cc6"
+web3 = Web3(Web3.HTTPProvider(infura_url))
+
+DOLOMITE_REST_ENDPOINT = "http://exchange-api.dolomite.io"
+
+
+
+def api_request_get(url):
+    
+        r = requests.get(url)
+        
+        print(r)
+        
+        response = r.json()
+        
+        return response
+    
+    
+def api_request_post(url, data):
+    
+        headers = {"Content-Type": "application/json"}
+    
+        r = requests.post(url, json=data, headers=headers)
+        
+        print(r)
+        
+        response = r.json()
+        
+        return response
+        
+        
+        
+def build_unsigned_order():
+        
+        
+        url = "%s/v1/orders/prepare" % (DOLOMITE_REST_ENDPOINT,)
+        
+        newAccount = web3.eth.account.create() #new auth address / private key 
+            
+            
+            
+        #Set fee amount for takers 
+        
+        fee_url = f"{DOLOMITE_REST_ENDPOINT}/v1/info"
+        
+        print(fee_url)
+        
+        res = api_request_get(fee_url)
+        
+        #res = self._api_request(http_method="get", url=fee_url)
+        
+        
+        
+        '''
+        #Gas fees 
+        
+        feeData = res["base_fee_amounts"] #find by quoteToken
+        
+        baseAmount = float(feeData[quoteToken]["amount"]) / math.pow(10, 18)
+        maxMatches = res["max_number_of_taker_matches_per_order"]
+        
+        takerGasFees = baseAmount * maxMatches
+        
+        
+        #Commission fees
+        
+        market_url = f"{DOLOMITE_REST_ENDPOINT}/v1/markets/{symbol}"
+        
+        res2 = api_request_get(market_url)
+        
+        #res2 = self._api_request(http_method="get", url=market_url)
+        
+        marketPrice = float(res2["current_price"]["amount"]) / math.pow(10, 18)
+        
+        takerRate = float(res["taker_fee_percentage"])
+        
+        commissionAmount = marketPrice * takerRate * float(amount)
```

Comment:
Check if `current_price` is 0. If it's 0, replace it with the order's price, since no market price exists